pkg/liqoctl/utils/pod/pod.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019-2024 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pod
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/kubectl/pkg/scheme"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/liqotech/liqo/pkg/liqoctl/test/network/setup"
+)
+
+// ExecInPod executes a command in a pod.
+func ExecInPod(ctx context.Context, clset *kubernetes.Clientset, cfg *rest.Config,
+	pod *corev1.Pod, cmd string) (stdout, stderr string, err error) {
+	// Prepare the API URL used to execute the command
+	url := clset.CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name(pod.Name).
+		Namespace(pod.Namespace).
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Command:   strings.Split(cmd, " "),
+			Container: pod.Spec.Containers[0].Name,
+			Stdin:     false,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       false,
+		}, scheme.ParameterCodec).URL()
+
+	// Execute the command
+	exec, err := remotecommand.NewSPDYExecutor(cfg, "POST", url)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to initialize command executor: %w", err)
+	}
+
+	// Capture the output and error streams
+	var stdoutBuff, stderrBuff bytes.Buffer
+	if err = exec.StreamWithContext(ctx, remotecommand.StreamOptions{
+		Stdin:  nil,
+		Stdout: &stdoutBuff,
+		Stderr: &stderrBuff,
+		Tty:    false,
+	}); err != nil {
+		return "", "", fmt.Errorf("failed to execute command: %w", err)
+	}
+
+	return stdoutBuff.String(), stderrBuff.String(), nil
+}
+
+// ListPods lists the pods.
+func ListPods(ctx context.Context, cl ctrlclient.Client, owner string, hostnetwork bool) (*corev1.PodList, error) {
+	deploymentName := setup.DeploymentName
+	if hostnetwork {
+		deploymentName = setup.DeploymentName + "-host"
+	}
+	pods := corev1.PodList{}
+	if err := cl.List(ctx, &pods,
+		ctrlclient.InNamespace(setup.NamespaceName),
+		ctrlclient.MatchingLabels{setup.PodLabelAppCluster: deploymentName + "-" + owner},
+	); err != nil {
+		return nil, fmt.Errorf("failed to list pods: %w", err)
+	}
+	return &pods, nil
+}
```

Comment:
This function is placed in a general util pkg but uses names from a specific network test. Is it possible to generalize it? if not I would place it inside the network/test folder